WorldGuard error on 1.16.2

There is an issue with WorldGuard on 1.16.2


```
[09:29:49] [Server thread/ERROR]: Could not pass event CraftReleaseEvent to Movecraft-Combat v1.0.0-SNAPSHOT.18
org.bukkit.event.EventException: null
        at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:319) ~[spigot-1.16.2.jar:git-Spigot-379750e-1e76d09]
        at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[spigot-1.16.2.jar:git-Spigot-379750e-1e76d09]
        at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:589) ~[spigot-1.16.2.jar:git-Spigot-379750e-1e76d09]
        at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:576) ~[spigot-1.16.2.jar:git-Spigot-379750e-1e76d09]
        at net.countercraft.movecraft.craft.CraftManager.removeCraft(CraftManager.java:170) ~[?:?]
        at net.countercraft.movecraft.sign.CraftSign.onSignClick(CraftSign.java:109) ~[?:?]
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_201]
        at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_201]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_201]
        at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_201]
        at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:315) ~[spigot-1.16.2.jar:git-Spigot-379750e-1e76d09]
        at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[spigot-1.16.2.jar:git-Spigot-379750e-1e76d09]
        at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:589) ~[spigot-1.16.2.jar:git-Spigot-379750e-1e76d09]
        at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:576) ~[spigot-1.16.2.jar:git-Spigot-379750e-1e76d09]
        at org.bukkit.craftbukkit.v1_16_R2.event.CraftEventFactory.callPlayerInteractEvent(CraftEventFactory.java:478) ~[spigot-1.16.2.jar:git-Spigot-379750e-1e76d09]
        at net.minecraft.server.v1_16_R2.PlayerInteractManager.a(PlayerInteractManager.java:444) ~[spigot-1.16.2.jar:git-Spigot-379750e-1e76d09]
        at net.minecraft.server.v1_16_R2.PlayerConnection.a(PlayerConnection.java:1278) ~[spigot-1.16.2.jar:git-Spigot-379750e-1e76d09]
        at net.minecraft.server.v1_16_R2.PacketPlayInUseItem.a(PacketPlayInUseItem.java:27) ~[spigot-1.16.2.jar:git-Spigot-379750e-1e76d09]
        at net.minecraft.server.v1_16_R2.PacketPlayInUseItem.a(PacketPlayInUseItem.java:1) ~[spigot-1.16.2.jar:git-Spigot-379750e-1e76d09]
        at net.minecraft.server.v1_16_R2.PlayerConnectionUtils.lambda$0(PlayerConnectionUtils.java:19) ~[spigot-1.16.2.jar:git-Spigot-379750e-1e76d09]
        at net.minecraft.server.v1_16_R2.TickTask.run(SourceFile:18) ~[spigot-1.16.2.jar:git-Spigot-379750e-1e76d09]
        at net.minecraft.server.v1_16_R2.IAsyncTaskHandler.executeTask(SourceFile:144) ~[spigot-1.16.2.jar:git-Spigot-379750e-1e76d09]
        at net.minecraft.server.v1_16_R2.IAsyncTaskHandlerReentrant.executeTask(SourceFile:23) ~[spigot-1.16.2.jar:git-Spigot-379750e-1e76d09]
        at net.minecraft.server.v1_16_R2.IAsyncTaskHandler.executeNext(SourceFile:118) ~[spigot-1.16.2.jar:git-Spigot-379750e-1e76d09]
        at net.minecraft.server.v1_16_R2.MinecraftServer.ba(MinecraftServer.java:941) ~[spigot-1.16.2.jar:git-Spigot-379750e-1e76d09]
        at net.minecraft.server.v1_16_R2.MinecraftServer.executeNext(MinecraftServer.java:934) ~[spigot-1.16.2.jar:git-Spigot-379750e-1e76d09]
        at net.minecraft.server.v1_16_R2.IAsyncTaskHandler.awaitTasks(SourceFile:127) ~[spigot-1.16.2.jar:git-Spigot-379750e-1e76d09]
        at net.minecraft.server.v1_16_R2.MinecraftServer.sleepForTick(MinecraftServer.java:918) ~[spigot-1.16.2.jar:git-Spigot-379750e-1e76d09]
        at net.minecraft.server.v1_16_R2.MinecraftServer.w(MinecraftServer.java:850) ~[spigot-1.16.2.jar:git-Spigot-379750e-1e76d09]
        at net.minecraft.server.v1_16_R2.MinecraftServer.lambda$0(MinecraftServer.java:164) ~[spigot-1.16.2.jar:git-Spigot-379750e-1e76d09]
        at java.lang.Thread.run(Unknown Source) [?:1.8.0_201]
Caused by: java.lang.NoSuchMethodError: com.sk89q.worldguard.bukkit.WorldGuardPlugin.getRegionManager(Lorg/bukkit/World;)Lcom/sk89q/worldguard/protection/managers/RegionManager;
        at net.countercraft.movecraft.combat.movecraftcombat.status.StatusManager.isInAirspace(StatusManager.java:156) ~[?:?]
        at net.countercraft.movecraft.combat.movecraftcombat.status.StatusManager.craftReleased(StatusManager.java:98) ~[?:?]
        at net.countercraft.movecraft.combat.movecraftcombat.listener.CraftReleaseListener.releaseListener(CraftReleaseListener.java:16) ~[?:?]
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_201]
        at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_201]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_201]
        at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_201]
        at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:315) ~[spigot-1.16.2.jar:git-Spigot-379750e-1e76d09]
        ... 30 more
```

Plugins:  LuckPerms, spark, void-generator, Vault, MassiveCore, Multiverse-Core, dynmap, Citizens, WorldEdit, Essentials, DtlPlayershops, Towny, WorldBorder, Cannons, WorldGuard, Factions, Dynmap-Factions, Movecraft, StructureBoxes, Factions3Chat, Periscopes, Movecraft-Combat, Movecraft-Factions

Here is with SNAPSHOT 35:

```
[15:06:02] [Server thread/ERROR]: Could not pass event CraftReleaseEvent to Movecraft-Combat v1.0.0-SNAPSHOT.35
org.bukkit.event.EventException: null
        at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:319) ~[spigot-1.16.2.jar:git-Spigot-379750e-1e76d09]
        at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[spigot-1.16.2.jar:git-Spigot-379750e-1e76d09]
        at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:589) ~[spigot-1.16.2.jar:git-Spigot-379750e-1e76d09]
        at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:576) ~[spigot-1.16.2.jar:git-Spigot-379750e-1e76d09]
        at net.countercraft.movecraft.craft.CraftManager.removeCraft(CraftManager.java:170) ~[?:?]
        at net.countercraft.movecraft.commands.ReleaseCommand.onCommand(ReleaseCommand.java:42) ~[?:?]
        at org.bukkit.command.PluginCommand.execute(PluginCommand.java:45) ~[spigot-1.16.2.jar:git-Spigot-379750e-1e76d09]
        at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:149) ~[spigot-1.16.2.jar:git-Spigot-379750e-1e76d09]
        at org.bukkit.craftbukkit.v1_16_R2.CraftServer.dispatchCommand(CraftServer.java:756) ~[spigot-1.16.2.jar:git-Spigot-379750e-1e76d09]
        at net.minecraft.server.v1_16_R2.PlayerConnection.handleCommand(PlayerConnection.java:1697) ~[spigot-1.16.2.jar:git-Spigot-379750e-1e76d09]
        at net.minecraft.server.v1_16_R2.PlayerConnection.a(PlayerConnection.java:1540) ~[spigot-1.16.2.jar:git-Spigot-379750e-1e76d09]
        at net.minecraft.server.v1_16_R2.PacketPlayInChat.a(PacketPlayInChat.java:47) ~[spigot-1.16.2.jar:git-Spigot-379750e-1e76d09]
        at net.minecraft.server.v1_16_R2.PacketPlayInChat.a(PacketPlayInChat.java:1) ~[spigot-1.16.2.jar:git-Spigot-379750e-1e76d09]
        at net.minecraft.server.v1_16_R2.PlayerConnectionUtils.lambda$0(PlayerConnectionUtils.java:19) ~[spigot-1.16.2.jar:git-Spigot-379750e-1e76d09]
        at net.minecraft.server.v1_16_R2.TickTask.run(SourceFile:18) ~[spigot-1.16.2.jar:git-Spigot-379750e-1e76d09]
        at net.minecraft.server.v1_16_R2.IAsyncTaskHandler.executeTask(SourceFile:144) ~[spigot-1.16.2.jar:git-Spigot-379750e-1e76d09]
        at net.minecraft.server.v1_16_R2.IAsyncTaskHandlerReentrant.executeTask(SourceFile:23) ~[spigot-1.16.2.jar:git-Spigot-379750e-1e76d09]
        at net.minecraft.server.v1_16_R2.IAsyncTaskHandler.executeNext(SourceFile:118) ~[spigot-1.16.2.jar:git-Spigot-379750e-1e76d09]
        at net.minecraft.server.v1_16_R2.MinecraftServer.ba(MinecraftServer.java:941) ~[spigot-1.16.2.jar:git-Spigot-379750e-1e76d09]
        at net.minecraft.server.v1_16_R2.MinecraftServer.executeNext(MinecraftServer.java:934) ~[spigot-1.16.2.jar:git-Spigot-379750e-1e76d09]
        at net.minecraft.server.v1_16_R2.IAsyncTaskHandler.awaitTasks(SourceFile:127) ~[spigot-1.16.2.jar:git-Spigot-379750e-1e76d09]
        at net.minecraft.server.v1_16_R2.MinecraftServer.sleepForTick(MinecraftServer.java:918) ~[spigot-1.16.2.jar:git-Spigot-379750e-1e76d09]
        at net.minecraft.server.v1_16_R2.MinecraftServer.w(MinecraftServer.java:850) ~[spigot-1.16.2.jar:git-Spigot-379750e-1e76d09]
        at net.minecraft.server.v1_16_R2.MinecraftServer.lambda$0(MinecraftServer.java:164) ~[spigot-1.16.2.jar:git-Spigot-379750e-1e76d09]
        at java.lang.Thread.run(Unknown Source) [?:1.8.0_201]
Caused by: java.lang.NoSuchMethodError: com.sk89q.worldguard.bukkit.WorldGuardPlugin.getRegionManager(Lorg/bukkit/World;)Lcom/sk89q/worldguard/protection/managers/RegionManager;
        at net.countercraft.movecraft.combat.movecraftcombat.status.StatusManager.isInAirspace(StatusManager.java:176) ~[?:?]
        at net.countercraft.movecraft.combat.movecraftcombat.status.StatusManager.craftReleased(StatusManager.java:98) ~[?:?]
        at net.countercraft.movecraft.combat.movecraftcombat.listener.CraftReleaseListener.releaseListener(CraftReleaseListener.java:16) ~[?:?]
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_201]
        at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_201]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_201]
        at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_201]
        at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:315) ~[spigot-1.16.2.jar:git-Spigot-379750e-1e76d09]
        ... 24 more
```

--- a/pom.xml
+++ b/pom.xml
@@ -61,7 +61,7 @@
         </repository>
         <repository>
             <id>sk89q-repo</id>
-            <url>http://maven.sk89q.com/repo/</url>
+            <url>https://maven.enginehub.org/repo/</url>
         </repository>
         <repository>
             <id>sonatype</id>
@@ -89,6 +89,16 @@
             <groupId>org.jetbrains</groupId>
             <artifactId>annotations-java5</artifactId>
             <version>RELEASE</version>
+        </dependency>
+        <dependency>
+            <groupId>com.sk89q.worldguard</groupId>
+            <artifactId>worldguard-core</artifactId>
+            <version>7.0.1-SNAPSHOT</version>
+        </dependency>
+        <dependency>
+            <groupId>com.sk89q.worldguard</groupId>
+            <artifactId>worldguard-bukkit</artifactId>
+            <version>7.0.1-SNAPSHOT</version>
         </dependency>
         <dependency>
             <groupId>com.sk89q</groupId>

--- a/src/main/java/net/countercraft/movecraft/combat/movecraftcombat/status/StatusManager.java
+++ b/src/main/java/net/countercraft/movecraft/combat/movecraftcombat/status/StatusManager.java
@@ -1,18 +1,26 @@
 package net.countercraft.movecraft.combat.movecraftcombat.status;
 
-import java.util.ArrayList;
-import java.util.Date;
-import java.util.HashMap;
-import java.util.HashSet;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.*;
+
+import com.sk89q.worldedit.bukkit.BukkitWorld;
+import com.sk89q.worldedit.math.BlockVector3;
+import com.sk89q.worldguard.WorldGuard;
+import com.sk89q.worldguard.bukkit.WorldGuardPlugin;
 import com.sk89q.worldguard.protection.ApplicableRegionSet;
-import com.sk89q.worldguard.protection.flags.DefaultFlag;
+import com.sk89q.worldguard.protection.flags.Flag;
+import com.sk89q.worldguard.protection.flags.Flags;
 import com.sk89q.worldguard.protection.flags.StateFlag;
+import com.sk89q.worldguard.protection.managers.RegionManager;
 import com.sk89q.worldguard.protection.regions.ProtectedRegion;
-import net.countercraft.movecraft.Movecraft;
 import net.countercraft.movecraft.MovecraftLocation;
 import net.countercraft.movecraft.combat.movecraftcombat.MovecraftCombat;
 import net.countercraft.movecraft.combat.movecraftcombat.localisation.I18nSupport;
-import net.countercraft.movecraft.craft.CraftManager;
+import net.countercraft.movecraft.combat.movecraftcombat.utils.LegacyUtils;
+import net.countercraft.movecraft.combat.movecraftcombat.utils.WorldGuard6Utils;
+import net.countercraft.movecraft.config.Settings;
 import net.countercraft.movecraft.utils.HitBox;
 import org.bukkit.*;
 import org.jetbrains.annotations.NotNull;
@@ -168,10 +176,14 @@ public class StatusManager extends BukkitRunnable {
         if(MovecraftCombat.getInstance().getWGPlugin() == null)
             return false;
 
+        if (LegacyUtils.getInstance().isLegacy()) {
+            return WorldGuard6Utils.isInAirspace(craft);
+        }
         for(MovecraftLocation l : getHitboxCorners(craft.getHitBox())) {
-            ApplicableRegionSet regions = MovecraftCombat.getInstance().getWGPlugin().getRegionManager(craft.getW()).getApplicableRegions(l.toBukkit(craft.getW()));
+            RegionManager manager = WorldGuard.getInstance().getPlatform().getRegionContainer().get(new BukkitWorld(craft.getW()));
+            ApplicableRegionSet regions = manager.getApplicableRegions(BlockVector3.at(l.getX(), l.getY(), l.getZ()));
             for(ProtectedRegion r : regions.getRegions()) {
-                if(r.getFlag(DefaultFlag.TNT) == StateFlag.State.DENY || r.getFlag(DefaultFlag.PVP) == StateFlag.State.DENY)
+                if(r.getFlag(Flags.TNT) == StateFlag.State.DENY || r.getFlag(Flags.PVP) == StateFlag.State.DENY)
                     return true;
             }
         }

--- a/src/main/java/net/countercraft/movecraft/combat/movecraftcombat/utils/WorldGuard6Utils.java
+++ b/src/main/java/net/countercraft/movecraft/combat/movecraftcombat/utils/WorldGuard6Utils.java
@@ -1,0 +1,65 @@
+package net.countercraft.movecraft.combat.movecraftcombat.utils;
+
+import com.sk89q.worldguard.bukkit.WorldGuardPlugin;
+import com.sk89q.worldguard.protection.ApplicableRegionSet;
+import com.sk89q.worldguard.protection.flags.DefaultFlag;
+import com.sk89q.worldguard.protection.flags.StateFlag;
+import com.sk89q.worldguard.protection.managers.RegionManager;
+import com.sk89q.worldguard.protection.regions.ProtectedRegion;
+import net.countercraft.movecraft.MovecraftLocation;
+import net.countercraft.movecraft.combat.movecraftcombat.MovecraftCombat;
+import net.countercraft.movecraft.craft.Craft;
+import net.countercraft.movecraft.utils.HitBox;
+import org.bukkit.Location;
+import org.bukkit.World;
+import org.jetbrains.annotations.NotNull;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+
+public class WorldGuard6Utils {
+
+    private static Method GET_REGION_MANAGER;
+    private static Method GET_APPLICABLE_REGIONS;
+
+    static {
+        try {
+            GET_REGION_MANAGER = WorldGuardPlugin.class.getDeclaredMethod("getRegionManager", World.class);
+            GET_APPLICABLE_REGIONS = RegionManager.class.getDeclaredMethod("getApplicableRegions", Location.class);
+        } catch (NoSuchMethodException e) {
+            e.printStackTrace();
+            GET_REGION_MANAGER = null;
+            GET_APPLICABLE_REGIONS = null;
+        }
+    }
+
+    private static ArrayList<MovecraftLocation> getHitboxCorners(@NotNull HitBox hitbox) {
+        ArrayList<MovecraftLocation> corners = new ArrayList<>();
+        corners.add(new MovecraftLocation(hitbox.getMinX(), hitbox.getMinY(), hitbox.getMinZ()));
+        corners.add(new MovecraftLocation(hitbox.getMinX(), hitbox.getMinY(), hitbox.getMaxZ()));
+        corners.add(new MovecraftLocation(hitbox.getMinX(), hitbox.getMaxY(), hitbox.getMinZ()));
+        corners.add(new MovecraftLocation(hitbox.getMinX(), hitbox.getMaxY(), hitbox.getMaxZ()));
+        corners.add(new MovecraftLocation(hitbox.getMaxX(), hitbox.getMinY(), hitbox.getMinZ()));
+        corners.add(new MovecraftLocation(hitbox.getMaxX(), hitbox.getMinY(), hitbox.getMaxZ()));
+        corners.add(new MovecraftLocation(hitbox.getMaxX(), hitbox.getMaxY(), hitbox.getMinZ()));
+        corners.add(new MovecraftLocation(hitbox.getMaxX(), hitbox.getMaxY(), hitbox.getMaxZ()));
+        return corners;
+    }
+
+    public static boolean isInAirspace(Craft craft) {
+        try {
+            for (MovecraftLocation l : getHitboxCorners(craft.getHitBox())) {
+                RegionManager manager = (RegionManager) GET_REGION_MANAGER.invoke(MovecraftCombat.getInstance().getWGPlugin(), craft.getW());
+                ApplicableRegionSet regions = (ApplicableRegionSet) GET_APPLICABLE_REGIONS.invoke(manager, l.toBukkit(craft.getW()));
+                for (ProtectedRegion r : regions.getRegions()) {
+                    if (r.getFlag(DefaultFlag.TNT) == StateFlag.State.DENY || r.getFlag(DefaultFlag.PVP) == StateFlag.State.DENY)
+                        return true;
+                }
+            }
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        return false;
+    }
+}

